icuSources/CMakeLists.txt

```diff
@@ -0,0 +1,519 @@
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the SwiftFoundation open source project
+##
+## Copyright (c) 2024 Apple Inc. and the SwiftFoundation project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.md for the list of SwiftFoundation project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+add_library(_FoundationICU
```

Comment:
would it be helpful to break the sources apart and add them in each subdirectory? You should be able to use `target_sources` to add sources to the target

```cmake
add_library(_FoundationICU)
target_include_directories(_FoundationICU PUBLIC include)
add_subdirectory(common)
add_subdirectory(i18n)
add_subdirectory(io)


target_link_libraries(_FoundationICU PRIVATE stdc++)
 target_compile_definitions(_FoundationICU PRIVATE U_I18N_IMPLEMENTATION U_IO_IMPLEMENTATION)
 if(CMAKE_SYSTEM_NAME STREQUAL Linux)
     target_compile_definitions(_FoundationICU PRIVATE SWIFT_PACKAGE=1)
 endif()
 set_target_properties(_FoundationICU PROPERTIES
     INSTALL_RPATH "$ORIGIN")

 # Install binary
 install(TARGETS _FoundationICU
     ARCHIVE DESTINATION lib/${install_directory}/${SWIFT_SYSTEM_NAME}
     LIBRARY DESTINATION lib/${install_directory}/${SWIFT_SYSTEM_NAME}
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})

 # Install headers
 install(DIRECTORY
             include/
         DESTINATION
             lib/${install_directory})

 set_property(GLOBAL APPEND PROPERTY SWIFT_FOUNDATION_ICU_EXPORTS _FoundationICU)
```

And then inside of each subdirectory, it would look something like this (I'll just do `io` because it's shorter, but it's the same pattern for each subdir).

```cmake
target_include_directories(_FoundationICU PRIVATE .)
target_sources(_FoundationICU
  PRIVATE
    locbund.cpp
    sprintf.cpp
    sscanf.cpp
    ucln_io.cpp
    ufile.cpp
    ufmt_cmn.cpp
    uprintf.cpp
    uprntf_p.cpp
    uscanf.cpp
    uscanf_p.cpp
    ustdio.cpp
    ustream.cpp)
```

I think that helps shrink this file down, and maybe gives us a prayer of keeping things in sync. Thoughts?

Reply:
Amazing suggestion. Thank you!